src/Symfony/Component/Console/Attribute/Argument.php

```diff
@@ -105,6 +119,16 @@ public function toInputArgument(): InputArgument
      */
     public function resolveValue(InputInterface $input): mixed
     {
-        return $input->getArgument($this->name);
+        $value = $input->getArgument($this->name);
+
+        if (is_subclass_of($this->typeName, \BackedEnum::class) && (is_string($value) || is_int($value))) {
+            try {
+                $value = ($this->typeName)::from($value);
+            } catch (\ValueError) {
+                throw InvalidArgumentException::fromEnumValue($this->name, $value, $this->suggestedValues);
+            }
```

Comment:
I used a try..catch here so that I keep the original `$value` in case of error. 
Using `BackedEnum::tryFrom($value)` could return `null`:

I don't like the alternative version:
```php
$enumValue = ($this->typeName)::tryFrom($value);
            if (null === $enumValue) {
                throw InvalidArgumentException::fromEnumValue($this->name, $value, $this->suggestedValues);
            }
            $value = $enumValue;
```